token_client.go

```diff
@@ -0,0 +1,80 @@
+package incognia
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type TokenClient struct {
+	ClientID      string
+	ClientSecret  string
+	netClient     *http.Client
+	tokenEndpoint string
+}
+
+type TokenClientConfig struct {
+	ClientID     string
+	ClientSecret string
+	Timeout      time.Duration
+}
+
+var (
+	ErrInvalidCredentials = errors.New("invalid credentials")
+)
+
+const (
+	tokenNetClientTimeout = 5 * time.Second
+)
```

Comment:
I would move both blocks to the beginning of the file, right after the import block, `const` first and `var` second. This a good practice to make easier to locate these global scopes. It only doesn't apply when defining enums.